### ISSUE TITLE
"--drive-shared-with-me" doesn't work on root directory for some commands

### ISSUE THREAD

**Issue by Dibbyo456** [rclone/rclone#3639]
#### Forum post:
https://forum.rclone.org/t/some-bugs-with-drive-shared-with-me-flag/12401

#### What is the problem you are having with rclone?

`--drive-shared-with-me` flag does not work as the way it should be when using without specifying any directory (Eg: `root`)

#### What is your rclone version (output from `rclone version`)

rclone v1.49.5
- os/arch: windows/amd64
- go version: go1.12.10

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Windows 10, 64 bit

####  Which cloud storage system are you using? (eg Google Drive)

Google Drive

**What exactly is happening:**

When using the following command
````
rclone ls gdriveP: -vv --drive-shared-with-me --fast-list
````
Rclone does not show the of "Shared with me" instead it shows the files of "My Drive". But if I specify a directory belongs to "Shared with me" then it works fine, eg:
````
rclone ls gdriveP:jordan -vv --drive-shared-with-me --fast-list
````

**Debug logs:**

**When using without directory (1st command):**
````
2019/10/18 17:02:56 DEBUG : rclone: Version "v1.49.5" starting with parameters ["rclone" "ls" "gdriveP:" "-vv" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log"]
2019/10/18 17:02:56 DEBUG : Using config file from "C:\\Users\\HARRY\\.config\\rclone\\rclone.conf"
2019/10/18 17:02:59 DEBUG : 9 go routines active
2019/10/18 17:02:59 DEBUG : rclone: Version "v1.49.5" finishing with parameters ["rclone" "ls" "gdriveP:" "-vv" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log"]
````

**When using with directory (2nd command):**
````
2019/10/18 17:10:33 DEBUG : rclone: Version "v1.49.5" starting with parameters ["rclone" "ls" "gdriveP:jordan" "-vv" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log"]
2019/10/18 17:10:33 DEBUG : Using config file from "C:\\Users\\HARRY\\.config\\rclone\\rclone.conf"
2019/10/18 17:10:36 DEBUG : 15 go routines active
2019/10/18 17:10:36 DEBUG : rclone: Version "v1.49.5" finishing with parameters ["rclone" "ls" "gdriveP:jordan" "-vv" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log"]
````

As you can there isn't much of a difference in the logs. The only difference I could see is 1st command has **9 go routines** active while the 2nd command  has **15 go routines** active.

**TL;DR:** `--drive-shared-with-me` is ignored when using without specifying any directory. This affects **copy**, **sync**, other commands too.

However there is some other things I need to mention. When using **lsd** command then everything works fine with/without directory..!! Eg:
````
rclone lsd gdriveP: --drive-shared-with-me --fast-list
````
In the above command the directories of "Shared with me" shows without any issue. In case if you're interested in the log, here it is:

````
2019/10/18 17:18:31 DEBUG : rclone: Version "v1.49.5" starting with parameters ["rclone" "lsd" "gdriveP:" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log" "-vv"]
2019/10/18 17:18:31 DEBUG : Using config file from "C:\\Users\\HARRY\\.config\\rclone\\rclone.conf"
2019/10/18 17:18:32 DEBUG : 5 go routines active
2019/10/18 17:18:32 DEBUG : rclone: Version "v1.49.5" finishing with parameters ["rclone" "lsd" "gdriveP:" "--drive-shared-with-me" "--fast-list" "--log-file" "rclone.log" "-vv"]
````

**TL;DR**: The described bug does not seem to apply on `lsd` and `lsf` commands.

**Comment by ncw** [comment]
Copied from the forum and updated:

`--drive-shared-with-me` seems to be broken... None of the files printed are shared...

```
$ rclone lsf -R TestDrive: --drive-shared-with-me=false --files-only | grep -v /
make_test_files.exe
README.md
z2.txt
z3.txt
make_test_files.exe
debian-8.6.0-amd64-netinst.iso
hello.txt
This is a Sheets.xlsx
^C
$ rclone lsf -R TestDrive: --drive-shared-with-me=false --files-only | grep -v /
make_test_files.exe
README.md
z2.txt
z3.txt
make_test_files.exe
debian-8.6.0-amd64-netinst.iso
hello.txt
This is a Sheets.xlsx
^C
$ rclone lsf TestDrive: --drive-shared-with-me --files-only
$ rclone lsf --disable ListR -R TestDrive: --drive-shared-with-me --files-only
```

With ListR we are showing all the files and without ListR we aren't showing any directories so we can recurse...

This might be related to this issue #3632

**Comment by ncw** [comment]
I've fixed this here - can you have a go?

https://beta.rclone.org/branch/v1.49.5-237-g60177d01-fix-3639-shared-with-me-beta/ (uploaded in 15-30 mins)

**Comment by Dibbyo456** [comment]
Yes, everything is fine now.

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50
